### PR TITLE
add proto message and gopaackage to fied types

### PR DESCRIPTION
**Pull request by ilackarms**
expand the autopilot model for `model.Field` from a single `string` to a struct with the following properties:

```go

type Type struct {
	// name of the type.
	Name string

	// proto message for the type, if the proto message is compiled with Autopilot
	Message proto.Message

	/*

		The go package containing the type of the file relative path to where the status and spec field can be found, if different than group root.
		Will be set automatically for proto-based types.

		If unset, AutoPilot uses the default types package for the type.
	*/
	GoPackage string
}

```

`RelativePathFromRoot` was removed from the `model.Group`, as `GoPackage` now provides the same functionality in the Type.

### CODE REVIEW

File: codegen/render/proto_types_renderer.go [hunk 1]
```diff
@@ -101,40 +98,35 @@ func (grp descriptorsWithGopath) getUniqueDescriptorsWithPath() []*model.Descrip
...
+
+		if pkgForGroup == "" {
+			// use default package
+			pkgForGroup = util.GoPackage(grp)
+			//// this case represents the local package, therefore no special pathing is necessary
+			//inputTmpls = inputTemplates{
+			//	protoDeepCopyTemplate: OutFile{
+			//		Path: protoDeepCopyGo,
+			//	},
+			//}
```
Comment (EItanya) [inline codegen/render/proto_types_renderer.go]: Commented code

File: codegen/model/resource.go [hunk 2]
```diff
@@ -61,18 +62,28 @@ type Resource struct {
...
+type Type struct {
+	// name of the type.
+	Name string
+
+	// proto message for the type, if the proto message is compiled with Autopilot
+	Message proto.Message
+
+	/*
+
+		The go package containing the type of the file relative path to where the status and spec field can be found, if different than group root.
```
Comment (EItanya) [inline codegen/model/resource.go]: relative path or go package?

File: codegen/cmd.go [hunk 3]
```diff
@@ -184,9 +190,35 @@ func (c Command) compileProtos(grp *render.Group) error {
...
+		grp.Resources[i] = resource
+	}
+
 	return nil
 }
 
+func addMessagesToResource(descriptors []*skmodel.DescriptorWithPath, resource *model.Resource) {
+	for _, fileDescriptor := range descriptors {
+
+		if fileDescriptor.GetPackage() == resource.Group.Group {
```
Comment (EItanya) [inline codegen/cmd.go:207]: Can't there potentially be many files in with the group's package?

File: codegen/model/resource.go (hunk 2 above)
Comment (ilackarms) [reply codegen/model/resource.go]: go package :+1

File: codegen/cmd.go (hunk 3 above)
Comment (ilackarms) [reply codegen/cmd.go:207]: fixed (this would overwrite with `nil` if there is more than 1 file and last file does not contain message)

**Review by EItanya — APPROVED** [review]

**Pull request merged**